.github/workflows/cannon.yml

```diff
@@ -0,0 +1,253 @@
+name: Cannon Tests
+
+env:
+  FOUNDRY_PROFILE: ci
+  OP_E2E_CANNON_ENABLED: "false"
+
+on:
+  push:
+    branches: [celestia-develop]
+  pull_request:
+    branches: [celestia-develop]
+  workflow_dispatch:
+
+jobs:
+  contracts-build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      # Debug step to check directory structure
+      - name: Debug directory structure
+        run: |
+          echo "Current directory: $(pwd)"
+          echo "Directory contents:"
+          ls -la
+          echo "Checking for packages directory:"
+          ls -la packages || echo "No packages directory"
+
+      # Install just command
+      - name: Install just
+        run: |
+          curl --proto '=https' --tlsv1.2 -sSf https://just.systems/install.sh | bash -s -- --to "$HOME/.local/bin"
+          echo "$HOME/.local/bin" >> "$GITHUB_PATH"
+
+      # Install Node.js without npm cache initially
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: "18"
+
+      - name: Install npm dependencies
+        run: |
+          if [ ! -d "packages/contracts-bedrock" ]; then
+            echo "contracts-bedrock directory not found. Creating it..."
+            mkdir -p packages/contracts-bedrock
+          fi
+          cd packages/contracts-bedrock
+
+          # Initialize package.json if it doesn't exist
+          if [ ! -f "package.json" ]; then
+            echo "Creating package.json..."
+            echo '{
+              "name": "contracts-bedrock",
+              "version": "1.0.0",
+              "private": true,
+              "description": "Optimism Bedrock smart contracts",
+              "scripts": {
+                "test": "echo \"Error: no test specified\" && exit 1"
+              }
+            }' > package.json
+          fi
```

Comment:
If these files and packages need to be present, why not just commit them? 
If it is an npm package, there should always be a package.json file. It doesn't really make sure to be creating that in CI.

Reply:
Yeah agree that would be more optimal, can change it later if this causes issues as this is all repurposed from the official OP CI workflow